rclcpp/include/rclcpp/experimental/buffers/intra_process_buffer.hpp

```diff
@@ -234,6 +234,54 @@ class TypedIntraProcessBuffer : public IntraProcessBuffer<MessageT, Alloc, Messa
   }
 };
 
+template<typename BufferT>
+class ServiceIntraProcessBuffer : public IntraProcessBufferBase
+{
+public:
+  RCLCPP_SMART_PTR_ALIASES_ONLY(ServiceIntraProcessBuffer)
+
+  virtual ~ServiceIntraProcessBuffer() {}
+
+  explicit
+  ServiceIntraProcessBuffer(
+    std::unique_ptr<BufferImplementationBase<BufferT>> buffer_impl)
+  {
+    buffer_ = std::move(buffer_impl);
+  }
+
+  // Not to be used in this class. Todo: review base class to avoid this.
+  bool use_take_shared_method() const override
+  {
+    throw std::runtime_error(
```

Comment:
do we really need to throw here?

Reply:
I'm just returning false now: [08290a8](https://github.com/ros2/rclcpp/pull/1847/commits/08290a80a9df30b0113e15426ab159004d412216)